test_conformance/images/kernel_read_write/test_write_image.cpp

```diff
@@ -448,38 +449,22 @@ int test_write_image( cl_device_id device, cl_context context, cl_command_queue
                     }
                     else if( imageInfo->format->image_channel_data_type == CL_HALF_FLOAT )
                     {
-
-                        // Compare half floats
-                        if( memcmp( resultBuffer, resultPtr, 2 * get_format_channel_count( imageInfo->format ) ) != 0 )
+                        cl_ushort *e = (cl_ushort *)resultBuffer;
+                        cl_ushort *a = (cl_ushort *)resultPtr;
+                        if( !validate_half_write_results( e, a, imageInfo ) )
                         {
-                            cl_ushort *e = (cl_ushort *)resultBuffer;
-                            cl_ushort *a = (cl_ushort *)resultPtr;
-                            int err_cnt = 0;
-
-                            //Fix up cases where we have NaNs
-                            for( size_t j = 0; j < get_format_channel_count( imageInfo->format ); j++ )
+                            totalErrors++;
+                            log_error( "ERROR: Sample %ld (%ld,%ld) did not validate! (%s)\n", i, x, y, mem_flag_names[ mem_flag_index ] );
+                            log_error( "    Expected: 0x%04x 0x%04x 0x%04x 0x%04x\n", e[ 0 ], e[ 1 ], e[ 2 ], e[ 3 ] );
+                            log_error( "    Actual:   0x%04x 0x%04x 0x%04x 0x%04x\n", a[ 0 ], a[ 1 ], a[ 2 ], a[ 3 ] );
+                            if( inputType == kFloat )
                             {
-                                if( is_half_nan( e[j] ) && is_half_nan(a[j]) )
-                                    continue;
-                                if( e[j] != a[j] )
-                                    err_cnt++;
-                            }
-
-                            if( err_cnt )
-                            {
-                                totalErrors++;
-                                log_error( "ERROR: Sample %ld (%ld,%ld) did not validate! (%s)\n", i, x, y, mem_flag_names[mem_flag_index] );
-                                log_error( "    Expected: 0x%04x 0x%04x 0x%04x 0x%04x\n", e[0], e[1], e[2], e[3] );
-                                log_error( "    Actual:   0x%04x 0x%04x 0x%04x 0x%04x\n", a[0], a[1], a[2], a[3] );
-                                if( inputType == kFloat )
-                                {
-                                    float *p = (float *)(char *)imagePtr;
-                                    log_error( "    Source: %a %a %a %a\n", p[ 0 ], p[ 1 ], p[ 2 ], p[ 3 ] );
-                                    log_error( "          : %12.24f %12.24f %12.24f %12.24f\n", p[ 0 ], p[ 1 ], p[ 2 ], p[ 3 ] );
-                                }
-                                if( ( --numTries ) == 0 )
-                                    return 1;
+                                float *p = (float *)(char *)imagePtr;
```

Comment:
seems like the `(char *)` is unecessary, could just be  `float *p = (float *)imagePtr`?